python/tvm/relax/op/nn/nn.py

```diff
@@ -622,3 +622,43 @@ def nll_loss(
       The computed result.
     """
     return _ffi_api.nll_loss(predictions, targets, weights, reduction, ignore_index)  # type: ignore
+
+
+def attention(
+    query: Expr, key: Expr, value: Expr, out_dtype: Optional[Union[str, DataType]] = None
+) -> Expr:
+    r"""Computes fused multi head attention.
+
+    All input tensors are of 4-D tensors with BSNH layout.
+
+    .. math::
+        FMA(Q, K, V) = \text{Softmax}(Q @ K^T) @ V
+
+    .. note::
+        The input tensor is required to have float16 dtype
```

Comment:
the requirement is backend specific and should be expressed in cutlass byoc instead of here